### PR TITLE
[MU4] Sessions manager

### PR DESCRIPTION
**Pull request by Eism**
Resolves: #10254

### CODE REVIEW

File: src/appshell/internal/appshellconfiguration.cpp [hunk 1]
```diff
@@ -238,3 +270,43 @@ std::string AppShellConfiguration::currentLanguageCode() const
 
     return locale.bcp47Name().toStdString();
 }
+
+mu::io::path AppShellConfiguration::sessionDataPath() const
+{
+    return globalConfiguration()->userAppDataPath() + "/session";
```
Comment (igorkorsukov) [inline src/appshell/internal/appshellconfiguration.cpp:276]: I think better to add extension (.json ?)

**Review by igorkorsukov — APPROVED** [review]

File: src/appshell/internal/appshellconfiguration.cpp (hunk 1 above)
Comment (Eism) [reply src/appshell/internal/appshellconfiguration.cpp:276]: It is a directory for session.
The extension for session is .json. See sessionFilePath and 
static const mu::io::path SESSION_FILE("/session.json");

**Comment by Jojo-Schmitz** [comment]
Why does MuseScore crash on generating the PNGs for the vtests?

**Comment by Eism** [comment]
> Why does MuseScore crash on generating the PNGs for the vtests?

I am working on it now

**Comment by DmitryArefiev** [comment]
Tested #10254 on Win10, Mac12, LinuxMint 20.2 - FIXED      

Please merge into master

**Pull request merged**